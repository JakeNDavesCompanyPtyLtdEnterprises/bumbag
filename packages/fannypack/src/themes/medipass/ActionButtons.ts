import { css, palette, theme, space } from '../../styled';

export default {
  base: css`
    flex-direction: row-reverse;
    justify-content: flex-end;
  `
};
