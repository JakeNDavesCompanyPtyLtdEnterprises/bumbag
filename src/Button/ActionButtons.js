// @flow
import React, { type Node } from 'react';

import type { Palette } from '../typesold';
import Button from './Button';
import Set from '../Set';

export type Props = {
  /** Custom button props for the cancel button */
  cancelProps?: Object,
  /** Custom text for the cancel button */
  cancelText?: string,
  children: Node,
  className?: string,
  /** Makes the submit button in a loading state */
  isLoading?: boolean,
  onClickSubmit?: Function,
  onClickCancel?: Function,
  /** Changes the color of the submit button */
  palette?: Palette,
  /** Custom button props for the submit button */
  submitProps?: Object,
  /** Custom text for the submit button */
  submitText?: string,
  /** Button type of the submit button */
  type?: string
};

const ActionButtons = ({
  cancelProps,
  cancelText,
  children,
  isLoading,
  onClickSubmit,
  onClickCancel,
  palette,
  submitProps,
  submitText,
  type,
  ...props
}: Props) => (
  <Set {...props}>
    <Button onClick={onClickCancel} {...cancelProps}>
      {/* $FlowFixMe */}
      {cancelText}
    </Button>
    <Button isLoading={isLoading} onClick={onClickSubmit} palette={palette} type={type} {...submitProps}>
      {/* $FlowFixMe */}
      {submitText}
    </Button>
  </Set>
);

ActionButtons.defaultProps = {
  cancelProps: {},
  cancelText: 'Cancel',
  className: undefined,
  isLoading: false,
  onClickSubmit: undefined,
  onClickCancel: undefined,
  palette: 'primary',
  submitProps: {},
  submitText: 'Submit',
  type: undefined
};

export default ActionButtons;
