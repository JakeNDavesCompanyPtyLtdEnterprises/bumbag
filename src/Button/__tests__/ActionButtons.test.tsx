import * as React from 'react';
import render from '../../_utils/tests/render';
import ActionButtons from '../ActionButtons';

it('renders correctly for default action buttons', () => {
  const { container } = render(<ActionButtons />);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly when a palette prop is set', () => {
  const { container } = render(<ActionButtons palette="text" />);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for action buttons with custom text', () => {
  const { container } = render(<ActionButtons cancelText="Close" submitText="Save" />);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for action buttons that is loading', () => {
  const { container } = render(<ActionButtons isLoading />);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for action buttons with a type', () => {
  const { container } = render(<ActionButtons type="submit" />);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for action buttons with custom props', () => {
  const { container } = render(
    <ActionButtons cancelText="Reject" submitProps={{ disabled: true }} cancelProps={{ palette: 'danger' }} />
  );
  expect(container.firstChild).toMatchSnapshot();
});
