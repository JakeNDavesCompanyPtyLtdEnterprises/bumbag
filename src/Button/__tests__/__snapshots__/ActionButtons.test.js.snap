// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for action buttons that is loading 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c5 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c8 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #3926a5;
  color: inherit;
}

.c11 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  color: #fff !important;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
  fill: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10 {
  font-size: 20px;
  -webkit-animation: iVXCSc 0.6s infinite linear;
  animation: iVXCSc 0.6s infinite linear;
  border: 0.1em solid #fff;
  border-radius: 100%;
  border-right-color: transparent;
  border-top-color: transparent;
  display: inline-block;
  height: 1em;
  width: 1em;
  position: relative;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #bfbfbf;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c4[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c4:hover {
  background-color: #efefef;
}

.c4:hover:active {
  background-color: #e3e3e3;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #3926a5;
  border: 1px solid #1c1352;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c7[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c7 {
  cursor: not-allowed;
}

.c7:focus {
  box-shadow: unset !important;
  outline: unset !important;
}

.c9 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 + span {
  opacity: 0;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-left: -0.5rem;
  margin-top: -0.5rem;
}

.c2 > * {
  margin-left: 0.5rem;
  margin-top: 0.5rem;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c1"
    spacing="xxsmall"
  >
    <button
      class="c4 c5"
      kind="default"
      style="size: default;"
    >
      <span
        class="c6"
      >
        Cancel
      </span>
    </button>
    <button
      class="c7 c8"
      kind="default"
      style="size: default;"
    >
      <div
        class="c9 c1"
      >
        <div
          class="c10 c11"
          color="primaryInverted"
          style="size: default;"
        />
      </div>
      <span
        class="c6"
      >
        Submit
      </span>
    </button>
  </div>
</div>
`;

exports[`renders correctly for action buttons with a type 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c5 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c8 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #3926a5;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
  fill: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #bfbfbf;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c4[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c4:hover {
  background-color: #efefef;
}

.c4:hover:active {
  background-color: #e3e3e3;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #3926a5;
  border: 1px solid #1c1352;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c7[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c7:hover {
  background-color: #322190;
}

.c7:hover:active {
  background-color: #2b1c7c;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-left: -0.5rem;
  margin-top: -0.5rem;
}

.c2 > * {
  margin-left: 0.5rem;
  margin-top: 0.5rem;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c1"
    spacing="xxsmall"
  >
    <button
      class="c4 c5"
      kind="default"
      style="size: default;"
    >
      <span
        class="c6"
      >
        Cancel
      </span>
    </button>
    <button
      class="c7 c8"
      kind="default"
      style="size: default;"
      type="submit"
    >
      <span
        class="c6"
      >
        Submit
      </span>
    </button>
  </div>
</div>
`;

exports[`renders correctly for action buttons with custom props 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c8 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #3926a5;
  color: inherit;
}

.c5 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #d60027;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
  fill: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #d60027;
  border: 1px solid #700014;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #e24c67 0px 0px 0px 2px;
}

.c4[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c4:hover {
  background-color: #bd0022;
}

.c4:hover:active {
  background-color: #a3001e;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #3926a5;
  border: 1px solid #1c1352;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c7[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-left: -0.5rem;
  margin-top: -0.5rem;
}

.c2 > * {
  margin-left: 0.5rem;
  margin-top: 0.5rem;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c1"
    spacing="xxsmall"
  >
    <button
      class="c4 c5"
      kind="default"
      style="size: default;"
    >
      <span
        class="c6"
      >
        Reject
      </span>
    </button>
    <button
      class="c7 c8"
      disabled=""
      kind="default"
      style="size: default;"
    >
      <span
        class="c6"
      >
        Submit
      </span>
    </button>
  </div>
</div>
`;

exports[`renders correctly for action buttons with custom text 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c5 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c8 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #3926a5;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
  fill: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #bfbfbf;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c4[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c4:hover {
  background-color: #efefef;
}

.c4:hover:active {
  background-color: #e3e3e3;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #3926a5;
  border: 1px solid #1c1352;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c7[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c7:hover {
  background-color: #322190;
}

.c7:hover:active {
  background-color: #2b1c7c;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-left: -0.5rem;
  margin-top: -0.5rem;
}

.c2 > * {
  margin-left: 0.5rem;
  margin-top: 0.5rem;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c1"
    spacing="xxsmall"
  >
    <button
      class="c4 c5"
      kind="default"
      style="size: default;"
    >
      <span
        class="c6"
      >
        Close
      </span>
    </button>
    <button
      class="c7 c8"
      kind="default"
      style="size: default;"
    >
      <span
        class="c6"
      >
        Save
      </span>
    </button>
  </div>
</div>
`;

exports[`renders correctly for default action buttons 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c5 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c8 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #3926a5;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
  fill: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #bfbfbf;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c4[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c4:hover {
  background-color: #efefef;
}

.c4:hover:active {
  background-color: #e3e3e3;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #3926a5;
  border: 1px solid #1c1352;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c7[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c7:hover {
  background-color: #322190;
}

.c7:hover:active {
  background-color: #2b1c7c;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-left: -0.5rem;
  margin-top: -0.5rem;
}

.c2 > * {
  margin-left: 0.5rem;
  margin-top: 0.5rem;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c1"
    spacing="xxsmall"
  >
    <button
      class="c4 c5"
      kind="default"
      style="size: default;"
    >
      <span
        class="c6"
      >
        Cancel
      </span>
    </button>
    <button
      class="c7 c8"
      kind="default"
      style="size: default;"
    >
      <span
        class="c6"
      >
        Submit
      </span>
    </button>
  </div>
</div>
`;

exports[`renders correctly when a palette prop is set 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c5 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c8 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #435a6f;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
  fill: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #bfbfbf;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c4[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c4:hover {
  background-color: #efefef;
}

.c4:hover:active {
  background-color: #e3e3e3;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #435a6f;
  border: 1px solid #1d262f;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #56748f 0px 0px 0px 2px;
}

.c7[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c7:hover {
  background-color: #394d5f;
}

.c7:hover:active {
  background-color: #30404f;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 {
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-left: -0.5rem;
  margin-top: -0.5rem;
}

.c2 > * {
  margin-left: 0.5rem;
  margin-top: 0.5rem;
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c1"
    spacing="xxsmall"
  >
    <button
      class="c4 c5"
      kind="default"
      style="size: default;"
    >
      <span
        class="c6"
      >
        Cancel
      </span>
    </button>
    <button
      class="c7 c8"
      kind="default"
      style="size: default;"
    >
      <span
        class="c6"
      >
        Submit
      </span>
    </button>
  </div>
</div>
`;
